parser/parser.go

```diff
@@ -147,6 +148,17 @@ func (p *parser) Parse() (res *Result, err error) {
 	}, nil
 }
 
+// encoreBuildContext creates a build context that mirrors what we pass onto the go compiler once the we trigger a build
+// of the application. This allows us to ignore `go` files which would be exlcuded during the build.
+//
+// For instance if a file has the directive `//go:build !encore` in it
+func encoreBuildContext() build.Context {
+	buildContext := build.Default
+	buildContext.ToolTags = append(buildContext.ToolTags, "encore")
```

Comment:
I've not set GOOS or GOARCH as the default context will pick that up for us, for now, I've just appended the `encore` tag like we do when we call out to `go build`. Although I've added it as a tool tag so users can still set build tags if wanted (build tags, tool tags and release tags are all combined when checking go build directives).

This means we also respect file name directives such as `api_windows.go` and `api_linux.go` now incase the same API needs difference implementations depending on OS.

Reply:
Ah nice. Does this make it easier to support e.g. `package foo_test` files? We have an outstanding bug to fix that.